services-camel/src/main/java/edu/unc/lib/dl/services/camel/destroyDerivatives/DestroyDerivativesProcessor.java

```diff
@@ -52,8 +53,10 @@ public DestroyDerivativesProcessor(String fileExtension, String derivativeBasePa
     public void process(Exchange exchange) {
         Message in = exchange.getIn();
         String binaryId = (String) in.getHeader(CdrBinaryPidId);
-        String binarySubPath = idToPath(binaryId, HASHED_PATH_DEPTH, HASHED_PATH_SIZE);
-        Path derivativePath = Paths.get(derivativeBasePath, binarySubPath, binaryId + fileExtension);
+        String binaryUUID = (String) in.getHeader(CdrBinaryUUID);
```

Comment:
I'd really prefer not to add an extra UUID field/header. We've been gradually phasing out `pid.getUuid()` in favor of `pid.getId()`, since the former just calls the latter, and neither necessarily returns a UUID. You can also get the ID/UUID portion back from the PID field, which it looks like you were doing before in DestroyedMsgProcessor. Was there a reason you needed to switch? I'd suggest:
```
String binaryPidId = (String) in.getHeader(CdrBinaryPidId);
PID binaryPid = PIDs.get(binaryPidId);
String binaryId = binaryPid.getId();
```

Reply:
IdToPath was trying to generate the derivative base path from /content/uuuid. So you'd end up with /co/nt/en etc, which I think is why it wasn't actually deleting things.